hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -733,22 +784,24 @@ protected Future createPutAllOperationFuture(final String name, MapEntries entri
         MapOperation op = operationProvider.createPutAllOperation(name, entries, false);
         op.setPartitionId(partitionId);
         final long size = entries.size();
-        final long time = System.currentTimeMillis();
+        final long time = getNow();
         InternalCompletableFuture<Object> f = operationService.invokeOnPartition(SERVICE_NAME, op, partitionId);
-        f.andThen(new ExecutionCallback() {
-            @Override
-            public void onResponse(Object response) {
-                LocalMapStatsProvider localMapStatsProvider = mapServiceContext.getLocalMapStatsProvider();
-                LocalMapStatsImpl localMapStats = localMapStatsProvider.getLocalMapStatsImpl(name);
-                long currentTime = System.currentTimeMillis();
-                localMapStats.incrementPuts(size, currentTime - time);
-            }
+        if (statisticsEnabled) {
+            f.andThen(new ExecutionCallback() {
+                @Override
+                public void onResponse(Object response) {
+                    LocalMapStatsProvider localMapStatsProvider = mapServiceContext.getLocalMapStatsProvider();
+                    LocalMapStatsImpl localMapStats = localMapStatsProvider.getLocalMapStatsImpl(name);
+                    long currentTime = getNow();
+                    localMapStats.updatePutStats(time - currentTime, size);
```

Comment:
isn't it sufficient to call `mapStatsCounter#updatePutStats` instead of all the code block in `onResponse` method?